### PR TITLE
Add projectId to bigquery table identifier

### PR DESCRIPTION
**Pull request by amine-hmila**
## Summary
Add projectId to bigquery table identifier

**Status: Ready to review**

**Breaking change? No**

## Contributor checklist:
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [ ] I have updated the Release notes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala [hunk 1]
```diff
@@ -56,7 +56,7 @@ trait BigQueryJobBase extends StrictLogging {
   val bigquery: BigQuery = BigQueryOptions.getDefaultInstance.getService
 
   val tableId: TableId = BigQueryJobBase.extractProjectDatasetAndTable(
-    cliConfig.outputDataset + "." + cliConfig.outputTable
+    projectId + ":" + cliConfig.outputDataset + "." + cliConfig.outputTable
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala:59]: cliConfig.outputDataset may already be in the form projected.datasetid

File: src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala [hunk 2]
```diff
@@ -68,13 +68,13 @@ trait BigQueryJobBase extends StrictLogging {
     }
   }
 
-  val bqTable = s"${cliConfig.outputDataset}.${cliConfig.outputTable}"
+  val bqTable = s"${projectId}:${cliConfig.outputDataset}.${cliConfig.outputTable}"
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala:71]: cliConfig.outputDataset may already be in the form projectid.datasetid

File: src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala [hunk 3]
```diff
@@ -68,13 +68,13 @@ trait BigQueryJobBase extends StrictLogging {
...
 
   def getOrCreateDataset(): Dataset = {
     val dataset = scala.Option(bigquery.getDataset(datasetId))
     dataset.getOrElse {
       val datasetInfo = DatasetInfo
-        .newBuilder(cliConfig.outputDataset)
+        .newBuilder(projectId, cliConfig.outputDataset)
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala:77]: cliConfig.outputDataset may already be in the form projectid.datasetid

File: src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala [hunk 4]
```diff
@@ -921,7 +921,8 @@ trait IngestionJob extends SparkJob {
             .getFields
             .size() == withScriptFieldsDF.schema.fields.length
         ) {
-          val bqTable = s"${domain.name}.${schema.name}"
+          val projectId = session.conf.get("fs.gs.project.id")
+          val bqTable = s"${projectId}:${domain.name}.${schema.name}"
```
Comment (hayssams) [inline src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala:925]: Ooh this is where we are having an issue.
At ingestion time we need to provide the project id. That what you do here.
What if the spark job is executed in a different project. In that case it won't work right ?

File: src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala (hunk 1 above)
Comment (amine-hmila) [reply src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala:59]: I tried adding the projectId in the yml domain file , 
```
  name: "${bq_project}:my_dataset"
  directory: "/my_dataset"
```
but i had this error 
```
ERROR com.ebiznext.comet.schema.handlers.SchemaHandler: Domain with name my-project-id:mydataset should respect the pattern [a-zA-Z][a-zA-Z0-9_]{1,100}
```

after adding the colons and dash to the validation regex it still seems not working as expected either 
` val dbNamePattern = Pattern.compile("[a-zA-Z:][a-zA-Z0-9_:/-]{1,100}")`

it expects to find the files under  `/datasets/pending/my-project-id:my_dataset` directory 
 
```
21/08/05 22:29:45 INFO com.ebiznext.comet.workflow.IngestionWorkflow: Watch Domain: my-project-id:my_dataset
21/08/05 22:29:45 INFO com.ebiznext.comet.workflow.IngestionWorkflow: List files in /datasets/pending/my-project-id:my_dataset
```

is that the right way to include the project id or do we still have some issues to fix ?

Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/job/index/bqload/BigQueryJobBase.scala:59]: You can't add a projected in the domain name
Actually you should not need to do that

**Comment by hayssams** [comment]
solved by referencing project id in  YML file

**Pull request closed without merging**